form-flow-valtimo/src/main/kotlin/com/ritense/valtimo/formflow/common/ValtimoFormFlow.kt

```diff
@@ -75,6 +88,97 @@ open class ValtimoFormFlow(
         taskService.complete(additionalProperties["taskInstanceId"] as String)
     }
 
+    /**
+     * Starts a new case by creating a document and starting a process.
+     */
+    @Transactional
+    open fun startCase(
+        formFlowInstanceId: FormFlowInstanceId,
+        submissionSavePath: Map<String, String>
+    ) {
+        val formFlowInstance = formFlowService.getInstanceById(formFlowInstanceId)
+        val documentDefinitionName = getRequiredAdditionalProperty(
+            formFlowInstance.getAdditionalProperties(),
+            "documentDefinitionName"
+        ).toString()
+
+        val submission = jacksonObjectMapper().readValue<JsonNode>(formFlowInstance.getSubmissionDataContext())
+
+        val submissionValues = submissionSavePath.entries.associate { it.key to getValue(submission, it.value) }
+        val submittedByType = valueResolverService.preProcessValuesForNewCase(submissionValues)
+
+        val document = documentService.createDocument(
+            NewDocumentRequest(
+                documentDefinitionName,
+                submittedByType["doc"] as JsonNode
+            )
+        ).also { result ->
+            if (result.errors().size > 0) {
+                throw RuntimeException(
+                    "Could not create document for document definition $documentDefinitionName\n" +
+                        "Reason:\n" +
+                        result.errors().joinToString(separator = "\n - ")
+                )
+            }
+        }.resultingDocument().orElseThrow()
+
+        val processDefinitionKey = getRequiredAdditionalProperty(formFlowInstance.getAdditionalProperties(), "processDefinitionKey").toString()
+        val startProcessForDocumentRequest = StartProcessForDocumentRequest(
+            document.id(),
+            processDefinitionKey,
+            submittedByType["pv"] as Map<String, Any>
+        )
+        val startProcessForDocumentResult = processDocumentService.startProcessForDocument(startProcessForDocumentRequest)
+        if (startProcessForDocumentResult.errors().isNotEmpty()) {
+            throw RuntimeException(
+                "Could not start process with definition $processDefinitionKey for document ${document.id()}\n" +
+                    "Reason:\n" +
+                    startProcessForDocumentResult.errors().joinToString(separator = "\n - ")
+            )
+        }
+    }
+
+    /**
+     * Starts a process for an existing case
+     */
+    @Transactional
+    open fun startSupportingProcess(
```

Comment:
Is it possible to combine this function with the `startCase(..)` function above? So that implementation teams don't have to create 2 identical form-flows that only differ in the `onComplete`